--- a/net/minecraft/world/level/block/state/BlockBehaviour.java
+++ b/net/minecraft/world/level/block/state/BlockBehaviour.java
@@ -84,7 +_,15 @@
 import net.minecraft.world.phys.shapes.VoxelShape;
 import org.jspecify.annotations.Nullable;
 
-public abstract class BlockBehaviour implements FeatureElement {
+// Fiddle start - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Allow delayed setting of key
+public abstract class BlockBehaviour implements FeatureElement, org.fiddlemc.fiddle.impl.moredatadriven.datapack.CopyResourcesFromDataPackRegistryToInternalRegistry.DelayedResourceKeyAttachable<Block> {
+    @Override
+    public void attachResourceKey(ResourceKey<Block> resourceKey) {
+        this.properties.setId(resourceKey);
+        this.drops = properties.effectiveDrops();
+        this.descriptionId = properties.effectiveDescriptionId();
+    }
+    // Fiddle end - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Allow delayed setting of key
     protected static final Direction[] UPDATE_SHAPE_ORDER = new Direction[]{
         Direction.WEST, Direction.EAST, Direction.NORTH, Direction.SOUTH, Direction.DOWN, Direction.UP
     };
@@ -98,13 +_,19 @@
     protected final boolean dynamicShape;
     protected final FeatureFlagSet requiredFeatures;
     protected final BlockBehaviour.Properties properties;
-    protected final Optional<ResourceKey<LootTable>> drops;
-    protected final String descriptionId;
+    // Fiddle start - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Allow delayed setting of key - Make dependent fields non-final
+    protected Optional<ResourceKey<LootTable>> drops;
+    protected String descriptionId;
+    // Fiddle end - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Allow delayed setting of key - Make dependent fields non-final
 
     public BlockBehaviour(BlockBehaviour.Properties properties) {
         this.hasCollision = properties.hasCollision;
+        // Fiddle start - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Allow delayed setting of key - Don't set dependent fields without a key
+        if (properties.id != null) {
         this.drops = properties.effectiveDrops();
         this.descriptionId = properties.effectiveDescriptionId();
+        }
+        // Fiddle end - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Allow delayed setting of key - Don't set dependent fields without a key
         this.explosionResistance = properties.explosionResistance;
         this.isRandomlyTicking = properties.isRandomlyTicking;
         this.soundType = properties.soundType;
@@ -123,7 +_,7 @@
     protected abstract MapCodec<? extends Block> codec();
 
     protected static <B extends Block> RecordCodecBuilder<B, BlockBehaviour.Properties> propertiesCodec() {
-        return BlockBehaviour.Properties.CODEC.fieldOf("properties").forGetter(BlockBehaviour::properties);
+        return org.fiddlemc.fiddle.impl.util.mojang.codec.CodecUtil.optionalFieldOf(BlockBehaviour.Properties.CODEC,"properties", BlockBehaviour.Properties::new).forGetter(BlockBehaviour::properties); // Fiddle - More data-driven - Minecraft registries - Types - Adapt existing block type registry - Make properties optional in block codec
     }
 
     public static <B extends Block> MapCodec<B> simpleCodec(Function<BlockBehaviour.Properties, B> factory) {
@@ -350,12 +_,20 @@
     }
 
     protected float getDestroyProgress(BlockState state, Player player, BlockGetter level, BlockPos pos) {
+        // Fiddle start - Block mapping - Breaking speed - Implement calculating destroy progress with given held item stack
+        return this.getDestroyProgress(state, player, level, pos, null);
+    }
+
+    protected float getDestroyProgress(BlockState state, Player player, BlockGetter level, BlockPos pos, @Nullable ItemStack heldItemStack) {
+        // Fiddle end - Block mapping - Breaking speed - Implement calculating destroy progress with given held item stack
         float destroySpeed = state.getDestroySpeed(level, pos);
         if (destroySpeed == -1.0F) {
             return 0.0F;
         } else {
-            int i = player.hasCorrectToolForDrops(state) ? 30 : 100;
-            return player.getDestroySpeed(state) / destroySpeed / i;
+            // Fiddle start - Block mapping - Breaking speed - Implement calculating destroy progress with given held item stack
+            int i = player.hasCorrectToolForDrops(state, heldItemStack) ? 30 : 100;
+            return player.getDestroySpeed(state, heldItemStack) / destroySpeed / i;
+            // Fiddle end - Block mapping - Breaking speed - Implement calculating destroy progress with given held item stack
         }
     }
 
@@ -434,11 +_,13 @@
         public final float destroySpeed;
         private final boolean requiresCorrectToolForDrops;
         private final boolean canOcclude;
-        private final BlockBehaviour.StatePredicate isRedstoneConductor;
-        private final BlockBehaviour.StatePredicate isSuffocating;
-        private final BlockBehaviour.StatePredicate isViewBlocking;
-        private final BlockBehaviour.StatePredicate hasPostProcess;
-        private final BlockBehaviour.StatePredicate emissiveRendering;
+        // Fiddle start - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
+        private final org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate isRedstoneConductor;
+        private final org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate isSuffocating;
+        private final org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate isViewBlocking;
+        private final org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate hasPostProcess;
+        private final org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate emissiveRendering;
+        // Fiddle end - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
         private final BlockBehaviour.@Nullable OffsetFunction offsetFunction;
         private final boolean spawnTerrainParticles;
         private final NoteBlockInstrument instrument;
@@ -525,7 +_,7 @@
         protected BlockStateBase(Block owner, Reference2ObjectArrayMap<Property<?>, Comparable<?>> values, MapCodec<BlockState> propertiesCodec) {
             super(owner, values, propertiesCodec);
             BlockBehaviour.Properties properties = owner.properties;
-            this.lightEmission = properties.lightEmission.applyAsInt(this.asState());
+            this.lightEmission = properties.lightEmission.apply(this.asState()); // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make light emission property serializable
             this.useShapeForLightOcclusion = owner.useShapeForLightOcclusion(this.asState());
             this.isAir = properties.isAir;
             this.ignitedByLava = properties.ignitedByLava;
@@ -754,7 +_,13 @@
         }
 
         public float getDestroyProgress(Player player, BlockGetter level, BlockPos pos) {
-            return this.getBlock().getDestroyProgress(this.asState(), player, level, pos);
+            // Fiddle start - Block mapping - Breaking speed - Implement calculating destroy progress with given held item stack
+            return this.getDestroyProgress(player, level, pos, null);
+        }
+
+        public float getDestroyProgress(Player player, BlockGetter level, BlockPos pos, @Nullable ItemStack heldItemStack) {
+            return this.getBlock().getDestroyProgress(this.asState(), player, level, pos, heldItemStack);
+            // Fiddle end - Block mapping - Breaking speed - Implement calculating destroy progress with given held item stack
         }
 
         public int getDirectSignal(BlockGetter level, BlockPos pos, Direction direction) {
@@ -1102,47 +_,50 @@
     }
 
     public static class Properties {
-        public static final Codec<BlockBehaviour.Properties> CODEC = MapCodec.unitCodec(() -> of());
-        Function<BlockState, MapColor> mapColor = state -> MapColor.NONE;
-        boolean hasCollision = true;
-        SoundType soundType = SoundType.STONE;
-        ToIntFunction<BlockState> lightEmission = state -> 0;
-        float explosionResistance;
-        float destroyTime;
-        boolean requiresCorrectToolForDrops;
-        boolean isRandomlyTicking;
-        float friction = 0.6F;
-        float speedFactor = 1.0F;
-        float jumpFactor = 1.0F;
+        public static final Codec<BlockBehaviour.Properties> CODEC = org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.BlockCodecs.PROPERTIES_CODEC; // Fiddle - More data-driven - Minecraft registries - Types - Adapt existing block type registry - Add missing properties codec implementation
+        // Fiddle start - More data-driven - Minecraft registries - Types - Properties serialization - Make properties fields public
+        public org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.BlockStateFunction<MapColor> mapColor = new org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.BlockStateFunction.Single<>(MapColor.NONE);
+        public boolean hasCollision = true;
+        public SoundType soundType = SoundType.STONE;
+        public org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.BlockStateFunction<Integer> lightEmission = new org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.BlockStateFunction.Single<>(0);
+        public boolean wasExplosionResistanceSet = false; // Fiddle - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Replace block arguments by references - Replace stairs explosion resistance implementation
+        public float explosionResistance;
+        public float destroyTime;
+        public boolean requiresCorrectToolForDrops;
+        public boolean isRandomlyTicking;
+        public float friction = 0.6F;
+        public float speedFactor = 1.0F;
+        public float jumpFactor = 1.0F;
         private @Nullable ResourceKey<Block> id;
         private DependantName<Block, Optional<ResourceKey<LootTable>>> drops = key -> Optional.of(
             ResourceKey.create(Registries.LOOT_TABLE, key.identifier().withPrefix("blocks/"))
         );
         private DependantName<Block, String> descriptionId = key -> Util.makeDescriptionId("block", key.identifier());
-        boolean canOcclude = true;
-        boolean isAir;
-        boolean ignitedByLava;
-        @Deprecated
-        boolean liquid;
-        @Deprecated
-        boolean forceSolidOff;
-        boolean forceSolidOn;
-        PushReaction pushReaction = PushReaction.NORMAL;
-        boolean spawnTerrainParticles = true;
-        NoteBlockInstrument instrument = NoteBlockInstrument.HARP;
-        boolean replaceable;
+        public boolean canOcclude = true;
+        public boolean isAir;
+        public boolean ignitedByLava;
+        @Deprecated
+        public boolean liquid;
+        @Deprecated
+        public boolean forceSolidOff;
+        public boolean forceSolidOn;
+        public PushReaction pushReaction = PushReaction.NORMAL;
+        public boolean spawnTerrainParticles = true;
+        public NoteBlockInstrument instrument = NoteBlockInstrument.HARP;
+        public boolean replaceable;
         BlockBehaviour.StateArgumentPredicate<EntityType<?>> isValidSpawn = (state, level, pos, random) -> state.isFaceSturdy(level, pos, Direction.UP)
             && state.getLightEmission() < 14;
-        BlockBehaviour.StatePredicate isRedstoneConductor = (state, level, pos) -> state.isCollisionShapeFullBlock(level, pos);
-        BlockBehaviour.StatePredicate isSuffocating = (state, level, pos) -> state.blocksMotion() && state.isCollisionShapeFullBlock(level, pos);
-        BlockBehaviour.StatePredicate isViewBlocking = this.isSuffocating;
-        BlockBehaviour.StatePredicate hasPostProcess = (state, level, pos) -> false;
-        BlockBehaviour.StatePredicate emissiveRendering = (state, level, pos) -> false;
-        boolean dynamicShape;
-        FeatureFlagSet requiredFeatures = FeatureFlags.VANILLA_SET;
+        public org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate isRedstoneConductor = org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate.IS_COLLISION_SHAPE_FULL_BLOCK;
+        public org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate isSuffocating = org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate.BLOCKS_MOTION_AND_IS_COLLISION_SHAPE_FULL_BLOCK;
+        public org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate isViewBlocking = this.isSuffocating;
+        public org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate hasPostProcess = org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate.NEVER;
+        public org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate emissiveRendering = org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate.NEVER;
+        public boolean dynamicShape;
+        public FeatureFlagSet requiredFeatures = FeatureFlags.VANILLA_SET;
         BlockBehaviour.@Nullable OffsetFunction offsetFunction;
+        // Fiddle end - More data-driven - Minecraft registries - Types - Properties serialization - Make properties fields public
 
-        private Properties() {
+        public Properties() { // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make properties constructor public
         }
 
         public static BlockBehaviour.Properties of() {
@@ -1168,7 +_,12 @@
             BlockBehaviour.Properties properties = new BlockBehaviour.Properties();
             BlockBehaviour.Properties properties1 = blockBehaviour.properties;
             properties.destroyTime = properties1.destroyTime;
+            // Fiddle start - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Replace block arguments by references - Replace stairs explosion resistance implementation
+            if (properties1.wasExplosionResistanceSet) {
             properties.explosionResistance = properties1.explosionResistance;
+            properties.wasExplosionResistanceSet = true;
+            }
+            // Fiddle end - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Replace block arguments by references - Replace stairs explosion resistance implementation
             properties.hasCollision = properties1.hasCollision;
             properties.isRandomlyTicking = properties1.isRandomlyTicking;
             properties.lightEmission = properties1.lightEmission;
@@ -1194,19 +_,25 @@
             return properties;
         }
 
+        // Fiddle start - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
+        public BlockBehaviour.Properties mapColor(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.BlockStateFunction<MapColor> mapColor) {
+            this.mapColor = mapColor;
+            return this;
+        }
+        // Fiddle end - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
+
         public BlockBehaviour.Properties mapColor(DyeColor mapColor) {
-            this.mapColor = state -> mapColor.getMapColor();
-            return this;
+            return this.mapColor(mapColor.getMapColor()); // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
         }
 
         public BlockBehaviour.Properties mapColor(MapColor mapColor) {
-            this.mapColor = state -> mapColor;
-            return this;
+            return this.mapColor(new org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.BlockStateFunction.Single<>(mapColor)); // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
         }
 
-        public BlockBehaviour.Properties mapColor(Function<BlockState, MapColor> mapColor) {
-            this.mapColor = mapColor;
-            return this;
+        // Fiddle start - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
+        public BlockBehaviour.Properties mapColor(List<Property<?>> properties, Function<org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.BlockStateFunction.ByProperties.PropertyValueMap, MapColor> byPropertyValues) {
+            return this.mapColor(new org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.BlockStateFunction.ByProperties<>(properties, byPropertyValues));
+            // Fiddle end - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
         }
 
         public BlockBehaviour.Properties noCollision() {
@@ -1240,10 +_,20 @@
             return this;
         }
 
-        public BlockBehaviour.Properties lightLevel(ToIntFunction<BlockState> lightEmission) {
-            this.lightEmission = lightEmission;
-            return this;
-        }
+        // Fiddle start - More data-driven - Minecraft registries - Types - Properties serialization - Make light emission property serializable
+        public BlockBehaviour.Properties lightLevel(int lightEmission) {
+            return this.lightLevel(new org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.BlockStateFunction.Single<>(lightEmission));
+        }
+
+        public BlockBehaviour.Properties lightLevel(Function<BlockState, Integer> lightEmission) {
+            if (lightEmission instanceof org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.BlockStateFunction<Integer> blockStateFunction) {
+                this.lightEmission = blockStateFunction;
+                return this;
+            }
+            // Assume that every piece of code that calls this method with an argument that is not a BlockStateFunction is calling it with a single value; for every other place we call the method with a BlockStateFunction instead
+            return this.lightLevel(lightEmission.apply(null));
+        }
+        // Fiddle end - More data-driven - Minecraft registries - Types - Properties serialization - Make light emission property serializable
 
         public BlockBehaviour.Properties strength(float destroyTime, float explosionResistance) {
             return this.destroyTime(destroyTime).explosionResistance(explosionResistance);
@@ -1319,26 +_,56 @@
         }
 
         public BlockBehaviour.Properties isRedstoneConductor(BlockBehaviour.StatePredicate isRedstoneConductor) {
+            // Fiddle start - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
+            return this.isRedstoneConductor(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate.wrap(isRedstoneConductor));
+        }
+
+        public BlockBehaviour.Properties isRedstoneConductor(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate isRedstoneConductor) {
+            // Fiddle end - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
             this.isRedstoneConductor = isRedstoneConductor;
             return this;
         }
 
         public BlockBehaviour.Properties isSuffocating(BlockBehaviour.StatePredicate isSuffocating) {
+            // Fiddle start - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
+            return this.isSuffocating(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate.wrap(isSuffocating));
+        }
+
+        public BlockBehaviour.Properties isSuffocating(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate isSuffocating) {
+            // Fiddle end - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
             this.isSuffocating = isSuffocating;
             return this;
         }
 
         public BlockBehaviour.Properties isViewBlocking(BlockBehaviour.StatePredicate isViewBlocking) {
+            // Fiddle start - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
+            return this.isViewBlocking(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate.wrap(isViewBlocking));
+        }
+
+        public BlockBehaviour.Properties isViewBlocking(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate isViewBlocking) {
+            // Fiddle end - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
             this.isViewBlocking = isViewBlocking;
             return this;
         }
 
         public BlockBehaviour.Properties hasPostProcess(BlockBehaviour.StatePredicate hasPostProcess) {
+            // Fiddle start - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
+            return this.hasPostProcess(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate.wrap(hasPostProcess));
+        }
+
+        public BlockBehaviour.Properties hasPostProcess(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate hasPostProcess) {
+            // Fiddle end - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
             this.hasPostProcess = hasPostProcess;
             return this;
         }
 
         public BlockBehaviour.Properties emissiveRendering(BlockBehaviour.StatePredicate emissiveRendering) {
+            // Fiddle start - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
+            return this.emissiveRendering(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate.wrap(emissiveRendering));
+        }
+
+        public BlockBehaviour.Properties emissiveRendering(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate emissiveRendering) {
+            // Fiddle end - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
             this.emissiveRendering = emissiveRendering;
             return this;
         }
@@ -1355,6 +_,7 @@
 
         public BlockBehaviour.Properties explosionResistance(float explosionResistance) {
             this.explosionResistance = Math.max(0.0F, explosionResistance);
+            this.wasExplosionResistanceSet = true; // Fiddle - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Replace block arguments by references - Replace stairs explosion resistance implementation
             return this;
         }
 
@@ -1423,7 +_,7 @@
     }
 
     @FunctionalInterface
-    public interface StatePredicate {
+    public interface StatePredicate extends java.io.Serializable { // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable - Make predicate extend serializable so that we can inspect it in KnownStatePrediate#wrap
         boolean test(BlockState state, BlockGetter level, BlockPos pos);
     }
 }
