--- a/src/main/java/io/papermc/paper/plugin/entrypoint/LaunchEntryPointHandler.java
+++ b/src/main/java/io/papermc/paper/plugin/entrypoint/LaunchEntryPointHandler.java
@@ -27,6 +_,7 @@
     // Utility
     public static void enterBootstrappers() {
         LaunchEntryPointHandler.INSTANCE.enter(Entrypoint.BOOTSTRAPPER);
+        org.fiddlemc.fiddle.impl.resourcepack.plugin.discover.FiddlePluginResourcePackDiscoveryImpl.get().fireComposeEvent(); // Fiddle - Resource pack - Plugin resource pack discovery - Implementation part - Fire event
     }
 
     @Override
@@ -37,6 +_,7 @@
         }
 
         storage.enter();
+        org.fiddlemc.fiddle.impl.packetmapping.item.namespacetooltip.NamespaceTooltips.registerNamespaceNames(storage); // Fiddle - Item mapping - Namespace tooltips - Register namespace
         this.enteredMap.put(entrypoint, true);
     }
 
