--- a/net/minecraft/world/level/block/Blocks.java
+++ b/net/minecraft/world/level/block/Blocks.java
@@ -44,11 +_,11 @@
 import net.minecraft.world.level.material.PushReaction;
 
 public class Blocks {
-    private static final BlockBehaviour.StatePredicate NOT_CLOSED_SHULKER = (state, level, pos) -> !(
+    public static final BlockBehaviour.StatePredicate NOT_CLOSED_SHULKER = (state, level, pos) -> !( // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable - Make public
             level.getBlockEntity(pos) instanceof ShulkerBoxBlockEntity shulkerBoxBlockEntity
         )
         || shulkerBoxBlockEntity.isClosed();
-    private static final BlockBehaviour.StatePredicate NOT_EXTENDED_PISTON = (state, level, pos) -> !state.getValue(PistonBaseBlock.EXTENDED);
+    public static final BlockBehaviour.StatePredicate NOT_EXTENDED_PISTON = (state, level, pos) -> !state.getValue(PistonBaseBlock.EXTENDED); // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable - Make public
     public static final Block AIR = register("air", AirBlock::new, BlockBehaviour.Properties.of().replaceable().noCollision().noLootTable().air());
     public static final Block STONE = register(
         "stone",
@@ -1404,7 +_,7 @@
         "wheat",
         CropBlock::new,
         BlockBehaviour.Properties.of()
-            .mapColor(blockState1 -> blockState1.getValue(CropBlock.AGE) >= 6 ? MapColor.COLOR_YELLOW : MapColor.PLANT)
+            .mapColor(java.util.List.of(CropBlock.AGE), blockState1 -> blockState1.getValue(CropBlock.AGE) >= 6 ? MapColor.COLOR_YELLOW : MapColor.PLANT) // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
             .noCollision()
             .randomTicks()
             .instabreak()
@@ -2062,7 +_,7 @@
             .strength(0.1F)
             .requiresCorrectToolForDrops()
             .sound(SoundType.SNOW)
-            .isViewBlocking((state, level, pos) -> state.getValue(SnowLayerBlock.LAYERS) >= 8)
+            .isViewBlocking(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate.MAX_SNOW_LAYERS) // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
             .pushReaction(PushReaction.DESTROY)
     );
     public static final Block ICE = register(
@@ -6288,7 +_,7 @@
             .strength(1.5F)
             .sound(SoundType.SCULK_SENSOR)
             .lightLevel(blockState1 -> 1)
-            .emissiveRendering((state, level, pos) -> SculkSensorBlock.getPhase(state) == SculkSensorPhase.ACTIVE)
+            .emissiveRendering(org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.KnownStatePredicate.SCULK_PHASE_ACTIVE)
     );
     public static final Block CALIBRATED_SCULK_SENSOR = register(
         "calibrated_sculk_sensor", CalibratedSculkSensorBlock::new, BlockBehaviour.Properties.ofLegacyCopy(SCULK_SENSOR)
@@ -7128,8 +_,10 @@
         return state -> state.getValue(BlockStateProperties.LIT) ? lightValue : 0;
     }
 
-    private static Function<BlockState, MapColor> waterloggedMapColor(MapColor unwaterloggedMapColor) {
-        return state -> state.getValue(BlockStateProperties.WATERLOGGED) ? MapColor.WATER : unwaterloggedMapColor;
+    // Fiddle start - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
+    private static org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.MapColorFunction waterloggedMapColor(MapColor unwaterloggedMapColor) {
+        return new org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.MapColorFunction.ByProperties(java.util.List.of(BlockStateProperties.WATERLOGGED), state -> state.getValue(BlockStateProperties.WATERLOGGED) ? MapColor.WATER : unwaterloggedMapColor);
+        // Fiddle end - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
     }
 
     private static Boolean never(BlockState state, BlockGetter level, BlockPos pos, EntityType<?> entity) {
@@ -7149,7 +_,7 @@
             name,
             properties -> new BedBlock(color, properties),
             BlockBehaviour.Properties.of()
-                .mapColor(blockState -> blockState.getValue(BedBlock.PART) == BedPart.FOOT ? color.getMapColor() : MapColor.WOOL)
+                .mapColor(java.util.List.of(BedBlock.PART), blockState -> blockState.getValue(BedBlock.PART) == BedPart.FOOT ? color.getMapColor() : MapColor.WOOL) // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
                 .sound(SoundType.WOOD)
                 .strength(0.2F)
                 .noOcclusion()
@@ -7160,7 +_,7 @@
 
     private static BlockBehaviour.Properties logProperties(MapColor sideColor, MapColor topColor, SoundType sound) {
         return BlockBehaviour.Properties.of()
-            .mapColor(state -> state.getValue(RotatedPillarBlock.AXIS) == Direction.Axis.Y ? sideColor : topColor)
+            .mapColor(java.util.List.of(RotatedPillarBlock.AXIS), state -> state.getValue(RotatedPillarBlock.AXIS) == Direction.Axis.Y ? sideColor : topColor) // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
             .instrument(NoteBlockInstrument.BASS)
             .strength(2.0F)
             .sound(sound)
@@ -7168,14 +_,14 @@
     }
 
     private static BlockBehaviour.Properties netherStemProperties(MapColor color) {
-        return BlockBehaviour.Properties.of().mapColor(blockState -> color).instrument(NoteBlockInstrument.BASS).strength(2.0F).sound(SoundType.STEM);
+        return BlockBehaviour.Properties.of().mapColor(color).instrument(NoteBlockInstrument.BASS).strength(2.0F).sound(SoundType.STEM); // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
     }
 
-    private static boolean always(BlockState state, BlockGetter level, BlockPos pos) {
+    public static boolean always(BlockState state, BlockGetter level, BlockPos pos) { // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable - Make public
         return true;
     }
 
-    private static boolean never(BlockState state, BlockGetter level, BlockPos pos) {
+    public static boolean never(BlockState state, BlockGetter level, BlockPos pos) { // Fiddle - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable - Make public
         return false;
     }
 
@@ -7271,6 +_,13 @@
 
     private static Block register(ResourceKey<Block> resourceKey, Function<BlockBehaviour.Properties, Block> factory, BlockBehaviour.Properties properties) {
         Block block = factory.apply(properties.setId(resourceKey));
+        // Fiddle start - More data-driven - Vanilla flag - Implementation part
+        block.markAsVanilla();
+        // Currently, there is no way to modify block states of vanilla blocks, so we assume all block states of vanilla blocks are vanilla too
+        for (BlockState blockState : block.getStateDefinition().getPossibleStates()) {
+            blockState.markAsVanilla();
+        }
+        // Fiddle end - More data-driven - Vanilla flag - Implementation part
         return Registry.register(BuiltInRegistries.BLOCK, resourceKey, block);
     }
 
@@ -7293,9 +_,10 @@
     static {
         for (Block block : BuiltInRegistries.BLOCK) {
             for (BlockState blockState : block.getStateDefinition().getPossibleStates()) {
-                Block.BLOCK_STATE_REGISTRY.add(blockState);
+                // Block.BLOCK_STATE_REGISTRY.add(blockState); // Fiddle - More data-driven - Minecraft registries - Register block states in BlockRegistry#register
                 blockState.initCache();
             }
         }
+        org.fiddlemc.fiddle.impl.moredatadriven.minecraft.BlockRegistry.get().canInitializeBlockStateCaches = true; // Fiddle - More data-driven - Minecraft registries - Initialize block state caches for non-vanilla blocks in BlockRegistry#register
     }
 }
