--- a/net/minecraft/world/level/block/BlockTypes.java
+++ b/net/minecraft/world/level/block/BlockTypes.java
@@ -9,9 +_,16 @@
 import net.minecraft.world.level.block.piston.PistonHeadBlock;
 
 public class BlockTypes {
-    public static final MapCodec<Block> CODEC = BuiltInRegistries.BLOCK_TYPE.byNameCodec().dispatchMap(Block::codec, Function.identity());
+    // Fiddle start - More data-driven - Minecraft registries - Types - Adapt existing block type registry
+    public static final MapCodec<Block> CODEC = BuiltInRegistries.FIDDLE_BLOCK_TYPE.byNameCodec().dispatchMap(block -> org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.WrappedBlockCodecImpl.wrap(block.codec()), org.fiddlemc.fiddle.api.moredatadriven.paper.registry.type.nms.WrappedBlockCodec::getCodec);
 
-    public static MapCodec<? extends Block> bootstrap(Registry<MapCodec<? extends Block>> registry) {
+    public static org.fiddlemc.fiddle.api.moredatadriven.paper.registry.type.nms.WrappedBlockCodec<?> bootstrap(Registry<org.fiddlemc.fiddle.api.moredatadriven.paper.registry.type.nms.WrappedBlockCodec<? extends net.minecraft.world.level.block.Block>> registry) {
+        class Registry {
+            static org.fiddlemc.fiddle.api.moredatadriven.paper.registry.type.nms.WrappedBlockCodec<?> register(net.minecraft.core.Registry<org.fiddlemc.fiddle.api.moredatadriven.paper.registry.type.nms.WrappedBlockCodec<? extends net.minecraft.world.level.block.Block>> registry, String name, MapCodec<? extends Block> value) {
+                return net.minecraft.core.Registry.register(registry, name, org.fiddlemc.fiddle.impl.moredatadriven.minecraft.type.WrappedBlockCodecImpl.wrap(value));
+            }
+        }
+        // Fiddle end - More data-driven - Minecraft registries - Types - Adapt existing block type registry
         Registry.register(registry, "block", Block.CODEC);
         Registry.register(registry, "air", AirBlock.CODEC);
         Registry.register(registry, "amethyst", AmethystBlock.CODEC);
@@ -213,7 +_,7 @@
         Registry.register(registry, "spore_blossom", SporeBlossomBlock.CODEC);
         Registry.register(registry, "stained_glass_pane", StainedGlassPaneBlock.CODEC);
         Registry.register(registry, "stained_glass", StainedGlassBlock.CODEC);
-        Registry.register(registry, "stair", StairBlock.CODEC);
+        Registry.register(registry, "stairs", StairBlock.CODEC); // Fiddle - More data-driven - Minecraft registries - Types - Adapt existing block type registry - Change "stair" to "stairs"
         Registry.register(registry, "standing_sign", StandingSignBlock.CODEC);
         Registry.register(registry, "stem", StemBlock.CODEC);
         Registry.register(registry, "stonecutter", StonecutterBlock.CODEC);
@@ -262,7 +_,7 @@
         Registry.register(registry, "weathering_copper_golem_statue", WeatheringCopperGolemStatueBlock.CODEC);
         Registry.register(registry, "weathering_copper_grate", WeatheringCopperGrateBlock.CODEC);
         Registry.register(registry, "weathering_copper_slab", WeatheringCopperSlabBlock.CODEC);
-        Registry.register(registry, "weathering_copper_stair", WeatheringCopperStairBlock.CODEC);
+        Registry.register(registry, "weathering_copper_stairs", WeatheringCopperStairBlock.CODEC); // Fiddle - More data-driven - Minecraft registries - Types - Adapt existing block type registry - Change "stair" to "stairs"
         Registry.register(registry, "weathering_copper_trap_door", WeatheringCopperTrapDoorBlock.CODEC);
         Registry.register(registry, "weathering_lantern", WeatheringLanternBlock.CODEC);
         Registry.register(registry, "weathering_lightning_rod", WeatheringLightningRodBlock.CODEC);
