--- a/net/minecraft/world/level/block/BrushableBlock.java
+++ b/net/minecraft/world/level/block/BrushableBlock.java
@@ -26,18 +_,10 @@
 import org.jspecify.annotations.Nullable;
 
 public class BrushableBlock extends BaseEntityBlock implements Fallable {
-    public static final MapCodec<BrushableBlock> CODEC = RecordCodecBuilder.mapCodec(
-        instance -> instance.group(
-                BuiltInRegistries.BLOCK.byNameCodec().fieldOf("turns_into").forGetter(BrushableBlock::getTurnsInto),
-                BuiltInRegistries.SOUND_EVENT.byNameCodec().fieldOf("brush_sound").forGetter(BrushableBlock::getBrushSound),
-                BuiltInRegistries.SOUND_EVENT.byNameCodec().fieldOf("brush_completed_sound").forGetter(BrushableBlock::getBrushCompletedSound),
-                propertiesCodec()
-            )
-            .apply(instance, BrushableBlock::new)
-    );
+    public static final MapCodec<BrushableBlock> CODEC = spout.common.moredatadriven.minecraft.type.BlockCodecs.brushableCodec(BrushableBlock::new); // Spout - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Initialize some references late
     private static final IntegerProperty DUSTED = BlockStateProperties.DUSTED;
     public static final int TICK_DELAY = 2;
-    private final Block turnsInto;
+    public Block turnsInto; // Spout - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Initialize some references late
     private final SoundEvent brushSound;
     private final SoundEvent brushCompletedSound;
 
