--- a/net/minecraft/world/level/block/WeatheringCopperStairBlock.java
+++ b/net/minecraft/world/level/block/WeatheringCopperStairBlock.java
@@ -9,14 +_,7 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public class WeatheringCopperStairBlock extends StairBlock implements WeatheringCopper {
-    public static final MapCodec<WeatheringCopperStairBlock> CODEC = RecordCodecBuilder.mapCodec(
-        instance -> instance.group(
-                WeatheringCopper.WeatherState.CODEC.fieldOf("weathering_state").forGetter(ChangeOverTimeBlock::getAge),
-                BlockState.CODEC.fieldOf("base_state").forGetter(weatheringCopperStairBlock -> weatheringCopperStairBlock.baseState),
-                propertiesCodec()
-            )
-            .apply(instance, WeatheringCopperStairBlock::new)
-    );
+    public static final MapCodec<WeatheringCopperStairBlock> CODEC = spout.common.moredatadriven.minecraft.type.BlockCodecs.weatheringCopperStairCodec(WeatheringCopperStairBlock::new); // Spout - More data-driven - Data pack - Copy loaded blocks and items to their internal registries - Initialize some references late
     private final WeatheringCopper.WeatherState weatherState;
 
     @Override
