--- a/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerLoginPacketListenerImpl.java
@@ -229,6 +_,7 @@
 
     private void finishLoginAndWaitForClient(GameProfile profile) {
         this.state = ServerLoginPacketListenerImpl.State.PROTOCOL_SWITCHING;
+        this.connection.clientViewSetter.sendClientModDetectionPacket(); // Fiddle - Client view - Set - Detect client mod
         this.connection.send(new ClientboundLoginFinishedPacket(profile));
         this.server.services().paper().filledProfileCache().add(profile); // Paper - update profile cache
     }
@@ -360,6 +_,7 @@
 
     @Override
     public void handleCustomQueryPacket(ServerboundCustomQueryAnswerPacket packet) {
+        if (this.connection.clientViewSetter.handleClientModDetectionPacket(packet)) return; // Fiddle - Client view - Set - Handle potential client mod response
         // Paper start - Add Velocity IP Forwarding Support
         try {
             this.handleCustomQueryPacket0(packet);
