--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlockStates.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlockStates.java
@@ -72,6 +_,7 @@
     private static final BlockStateFactory<?> DEFAULT_FACTORY = new BlockStateFactory<>(CraftBlockState.class) {
         @Override
         public CraftBlockState createBlockState(World world, BlockPos pos, net.minecraft.world.level.block.state.BlockState state, BlockEntity blockEntity) {
+            if (blockEntity != null && !blockEntity.getType().isVanilla()) return new CraftBlockState(world, pos, state); // Fiddle - More data-driven - Block entities - TODO provide API so that plugins can read data from custom block entities using the API. For now, return a generic CraftBlockState instead of erroring.
             // Paper - revert upstream's revert of the block state changes. Block entities that have already had the block type set to AIR are still valid, upstream decided to ignore them
             Preconditions.checkState(blockEntity == null, "Unexpected BlockState for %s", CraftBlockType.minecraftToBukkit(state.getBlock()));
             return new CraftBlockState(world, pos, state);
