--- a/net/minecraft/core/MappedRegistry.java
+++ b/net/minecraft/core/MappedRegistry.java
@@ -40,7 +_,7 @@
     private Lifecycle registryLifecycle;
     private final Map<TagKey<T>, HolderSet.Named<T>> frozenTags = new IdentityHashMap<>();
     MappedRegistry.TagSet<T> allTags = MappedRegistry.TagSet.unbound();
-    private boolean frozen;
+    boolean frozen;
     private @Nullable Map<T, Holder.Reference<T>> unregisteredIntrusiveHolders;
     // Paper start - support pre-filling in registry mod API
     private final Map<Identifier, T> temporaryUnfrozenMap = new HashMap<>();
@@ -287,9 +_,20 @@
 
     @Override
     public Registry<T> freeze() {
+        // Spout start - More data-driven - Data pack - Delayed frozen registries - Delay freezing registries
+        if (!spout.server.paper.impl.moredatadriven.datapack.delayedfrozenregistries.DelayedRegistryFreezing.isFreezingAllowed(this)) {
+            spout.server.paper.impl.moredatadriven.datapack.delayedfrozenregistries.DelayedRegistryFreezing.freezeLater(this);
+            return this;
+        }
         if (this.frozen) {
             return this;
         } else {
+            return this.actuallyFreeze();
+        }
+    }
+
+    protected Registry<T> actuallyFreeze() {
+        // Spout end - More data-driven - Data pack - Delayed frozen registries - Delay freezing registries
             this.frozen = true;
             this.temporaryUnfrozenMap.clear(); // Paper - support pre-filling in registry mod API
             this.byValue.forEach((object, reference) -> reference.bindValue((T)object));
@@ -330,7 +_,7 @@
                     }
                 }
             }
-        }
+        // } // Spout - More data-driven - Data pack - Delayed frozen registries - Delay freezing registries
     }
 
     @Override
