--- a/net/minecraft/core/MappedRegistry.java
+++ b/net/minecraft/core/MappedRegistry.java
@@ -40,7 +_,7 @@
     private Lifecycle registryLifecycle;
     private final Map<TagKey<T>, HolderSet.Named<T>> frozenTags = new IdentityHashMap<>();
     MappedRegistry.TagSet<T> allTags = MappedRegistry.TagSet.unbound();
-    private boolean frozen;
+    boolean frozen;
     private @Nullable Map<T, Holder.Reference<T>> unregisteredIntrusiveHolders;
     // Paper start - support pre-filling in registry mod API
     private final Map<Identifier, T> temporaryUnfrozenMap = new HashMap<>();
@@ -287,6 +_,12 @@
 
     @Override
     public Registry<T> freeze() {
+        // Fiddle start - More data-driven - Data pack - Delayed frozen registries - Delay freezing registries
+        if (!org.fiddlemc.fiddle.impl.moredatadriven.datapack.delayedfrozenregistries.DelayedRegistryFreezing.isFreezingAllowed(this)) {
+            org.fiddlemc.fiddle.impl.moredatadriven.datapack.delayedfrozenregistries.DelayedRegistryFreezing.freezeLater(this);
+            return this;
+        }
+        // Fiddle end - More data-driven - Data pack - Delayed frozen registries - Delay freezing registries
         if (this.frozen) {
             return this;
         } else {
