--- a/net/minecraft/server/level/ServerEntity.java
+++ b/net/minecraft/server/level/ServerEntity.java
@@ -330,12 +_,14 @@
         if (this.entity instanceof LivingEntity livingEntity) {
             Collection<AttributeInstance> syncableAttributes = livingEntity.getAttributes().getSyncableAttributes();
             // CraftBukkit start - If sending own attributes send scaled health instead of current maximum health
+            @Nullable ServerPlayer selfPlayer = null; // Fiddle - Block mapping - Breaking speed - Pass self player to attributes packet
             if (this.entity.getId() == player.getId()) {
+                selfPlayer = player; // Fiddle - Block mapping - Breaking speed - Pass self player to attributes packet
                 ((ServerPlayer) this.entity).getBukkitEntity().injectScaledMaxHealth(syncableAttributes, false);
             }
             // CraftBukkit end
             if (!syncableAttributes.isEmpty()) {
-                consumer.accept(new ClientboundUpdateAttributesPacket(this.entity.getId(), syncableAttributes));
+                consumer.accept(new ClientboundUpdateAttributesPacket(this.entity.getId(), syncableAttributes, selfPlayer)); // Fiddle - Block mapping - Breaking speed - Pass self player to attributes packet
             }
         }
 
@@ -404,7 +_,12 @@
                     serverPlayer.getBukkitEntity().injectScaledMaxHealth(attributesToSync, false);
                 }
                 // CraftBukkit end
-                this.synchronizer.sendToTrackingPlayersAndSelf(new ClientboundUpdateAttributesPacket(this.entity.getId(), attributesToSync));
+                // Fiddle start - Block mapping - Breaking speed - Pass self player to attributes packet
+                this.synchronizer.sendToTrackingPlayers(new ClientboundUpdateAttributesPacket(this.entity.getId(), attributesToSync, null));
+                if (this.entity instanceof ServerPlayer serverPlayer) {
+                    serverPlayer.connection.send(new net.minecraft.network.protocol.game.ClientboundUpdateAttributesPacket(this.entity.getId(), attributesToSync, serverPlayer));
+                }
+                // Fiddle ebd - Block mapping - Breaking speed - Pass self player to attributes packet
             }
 
             attributesToSync.clear();
